File: cpp/11047.cpp
#include <iostream>
using namespace std;
int main(){
    int n = 0;
    int k = 0;
    int count = 0;
    int num_list[11];
    cin >> n >> k;
    for(int i=0;i<n;i++){
        cin >> num_list[i];
    }
    for(int j=n-1;j>=0;j--){
        if (num_list[j] <= k){
            count += k/num_list[j];
            k = k%num_list[j];
        }
            
    }
    cout << count;
    return 0;
}

File: python/11047.py
import sys
n, k = map(int, input().split())
count = 0
num_list = [int(input()) for _ in range(n)]
m=0
num_list = list(reversed(num_list))
for j in range(n):
    if num_list[j] <= k :
        count += k//num_list[j]
        k = k%num_list[j]
print(count)